sdks/uniswapx-sdk/src/builder/V3DutchOrderBuilder.ts

```diff
@@ -0,0 +1,273 @@
+        );
+        invariant(
+            this.info.cosignerData.outputOverrides.length > 0,
+            "outputOverrides not set"
+        );
+        this.info.cosignerData.outputOverrides.forEach((override, idx) => {
+            invariant(
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                override.gte(this.info.outputs![idx].startAmount),
+                "outputOverride smaller than original output"
+            );
+        });
+        invariant(this.info.input !== undefined, "original input not set");
+        // We are not checking if the decayStartTime is before the deadline because it is not enforced in the smart contract
+
+        return new CosignedV3DutchOrder(
+            Object.assign(this.getOrderInfo(), {
+                cosignerData: this.info.cosignerData,
+                input: this.info.input,
+                outputs: this.info.outputs,
+                cosigner: this.info.cosigner,
+                cosignature: this.info.cosignature,
+            }),
+            this.chainId,
+            this.permit2Address
+        );
+
+    }
+    private permit2Address: string;
+    private info: Partial<CosignedV3DutchOrderInfo>;
+
+    constructor(
+        private chainId: number,
+        reactorAddress?: string,
+        _permit2Address?: string
+    ) {
+        super();
+
+        this.reactor(getReactor(chainId, OrderType.Dutch_V3, reactorAddress));
+        this.permit2Address = getPermit2(chainId, _permit2Address);
+
+        this.info = {
+            outputs: [],
+            cosignerData: {
+                decayStartBlock: 0,
+                exclusiveFiller: ethers.constants.AddressZero,
+                exclusivityOverrideBps: BigNumber.from(0),
+                inputOverride: BigNumber.from(0),
+                outputOverrides: [],
+            },
+        };
+    }
+
+    cosigner(cosigner: string): this {
+        this.info.cosigner = cosigner;
+        return this;
+    }
+
+    cosignature(cosignature: string | undefined): this {
+        this.info.cosignature = cosignature;
+        return this;
+    }
+
+    decayStartBlock(decayStartBlock: number): this {
+        if (!this.info.cosignerData) {
+            this.initializeCosignerData({ decayStartBlock });
+        } else {
+            this.info.cosignerData.decayStartBlock = decayStartBlock;
+        }
+        return this;
+    }
+
+    private initializeCosignerData(data: Partial<V3CosignerData>): void {
+        this.info.cosignerData = {
+          decayStartBlock: 0,
+          exclusiveFiller: ethers.constants.AddressZero,
+          exclusivityOverrideBps: BigNumber.from(0),
+          inputOverride: BigNumber.from(0),
+          outputOverrides: [],
+          ...data,
+        };
+    }
+
+    input(input: V3DutchInput): this {
+        this.info.input = input;
+        return this;
+    }
+
+    output(output: V3DutchOutput): this {
+        invariant(
```

Comment:
flagging this in case you want to remove it

Reply:
removed and added test for it, ty!